### PR TITLE
Improve state synthesis tests for remote simulators and fix flakes

### PR DESCRIPTION
**Pull request by annagrin**
Fix flaky tests introduced in previous PRs (#2212 and #2211)

### CODE REVIEW

File: targettests/Remote-Sim/qvector_init_from_state.cpp [hunk 1]
```diff
@@ -31,6 +31,7 @@ __qpu__ void test_init_large_state() {
 
 __qpu__ void test_state_param(cudaq::state* state) {
   cudaq::qvector q1(state);
+  x(q1);
```
Comment (khalatepradnya) [inline targettests/Remote-Sim/qvector_init_from_state.cpp:34]: Q: Does addition of this operation make the test deterministic?

File: targettests/Remote-Sim/qvector_init_from_state.cpp [hunk 2]
```diff
@@ -77,68 +78,57 @@ int main() {
...
   }
-
 // CHECK: Passing state created from data as argument (kernel mode)
-// CHECK: 000
-// CHECK: 100
-
 // CHECK: 011
 // CHECK: 111
 
+// CHECK: 000
```
Comment (khalatepradnya) [inline targettests/Remote-Sim/qvector_init_from_state.cpp:85]: Q: Do we need `CHECK-DAG` directive if the output order varies?

**Review by khalatepradnya — APPROVED** [review]
LGTM. 
Minor questions.

File: targettests/Remote-Sim/qvector_init_from_state.cpp (hunk 1 above)
Comment (annagrin) [reply targettests/Remote-Sim/qvector_init_from_state.cpp:34]: No, this part is just for making the test better - if it changes the output every time it's called we can test that iterations fip the qubit values and not just get stale values.

File: targettests/Remote-Sim/qvector_init_from_state.cpp (hunk 2 above)
Comment (annagrin) [reply targettests/Remote-Sim/qvector_init_from_state.cpp:85]: The output order should not vary, the change in the output is due to the addition of the gate x() in `test_state_param`. Apologies for confusion, I'll change the title of the PR to "improve tests and remove flakes"

**Pull request merged**